bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -322,6 +282,120 @@ private void scan() {
         logger.trace("IpAddonFinder::scan completed");
     }
 
+    private void scanBroadcast(AddonInfo candidate, String request, String response, int timeoutMs, int destPort) {
+        if (request.isEmpty()) {
+            logger.warn("{}: match-property request \"{}\" is unknown", candidate.getUID(), TYPE_IP_BROADCAST);
+            return;
+        }
+        if (response.isEmpty()) {
+            logger.warn("{}: match-property response \"{}\" is unknown", candidate.getUID(), TYPE_IP_BROADCAST);
+            return;
+        }
+        try (DatagramSocket socket = new DatagramSocket()) {
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                @Nullable
+                NetworkInterface networkInterface = interfaces.nextElement();
+                if (networkInterface.isLoopback() || !networkInterface.isUp()) {
+                    continue;
+                }
+                for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                    if (interfaceAddress.getBroadcast() == null) {
+                        continue;
+                    }
+                    socket.setBroadcast(true);
+                    socket.setSoTimeout(timeoutMs);
+                    byte[] sendBuffer = buildByteArray(request);
+                    DatagramPacket sendPacket = new DatagramPacket(sendBuffer, sendBuffer.length,
+                            interfaceAddress.getBroadcast(), destPort);
```

Comment:
Rather than using the standard broadcast address, you could use the configurable one from network config. This could limit network traffic.

Reply:
I had a look at `NetUtil.getConfiguredBroadcastAddress()`. If I understand it correctly, it will return the first broadcast address for any interface, if not explicitly configured and no primary IP address has been configured either:
https://github.com/openhab/openhab-core/blob/875ebaaef7eba2420469ba57c44dbc42eee19d76/bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java#L129-L135

and:
https://github.com/openhab/openhab-core/blob/875ebaaef7eba2420469ba57c44dbc42eee19d76/bundles/org.openhab.core/src/main/java/org/openhab/core/net/NetUtil.java#L286-L302

In the case of a new installation, do you think this network configuration is in place? I just checked my production system, and even here, I don't have neither of them configured:

![image](https://github.com/openhab/openhab-core/assets/19519842/06149fa9-2f46-4b50-8302-1501771017db)

My test instance looks similar:

![image](https://github.com/openhab/openhab-core/assets/19519842/bbcdca17-ef6f-4066-8dc9-4bd4f7d9258c)

So if you are referring to using `NetUtil.getConfiguredBroadcastAddress()`, my question is: Won't this limit the number of potential results (while also limiting the network traffic, which of course is a good thing)?

Reply:
Yes, indeed, it will limit it. If nothing is configured, it still has defaults set in the service initialization. But I would think we would want to be able to influence this in this finder. Rather than creating another configuration option, I believe we should use what is already there.
See also https://github.com/openhab/openhab-core/issues/4047. I believe there are too many places where configuration parameters for these are created ad-hoc, ultimately creating complexity and confusion.

Reply:
What would be your advise for this PR? I don't see how `NetUtil` can be used, since it would artificially/randomly limit the number of results. So I'm not sure how to proceed with this comment.

Reply:
I don't think we should create something that does an IP broadcast on all interfaces by default. We see too many discussions in the forum (some right, some wrong) about how these finders hog up (network) resources. IP broadcast on all interfaces by default would be a no go in my opinion. NetUtil offers a way to restrict it to a single subnet (which would be what would normally be required) with a default and a standard way of configuring it. If you want to have the ability to broadcast on all interfaces, one way could be to iterate over interfaces with the interface broadcast address if the broadcast address is set to 255.255.255.255, but I actually don't think that is necessary. In any way, it would be a configuration decision and not a default.

Reply:
So your suggestion is to use `NetUtil.getConfiguredBroadcastAddress()`?

Reply:
Yes, indeed. My reasoning is:
- Have some limit on what is sent by default.
- Have it configurable without creating yet another way of configuring network settings.

Reply:
Okay, I've implemented that.

Already kind of mentioned, but just to be clear: My Danfoss Air can no longer be found from my test system, since the broadcast address was defaulted to 192.168.129.255 (from the screenshot above). After configuring 192.168.0.255 and restarting openHAB, it was found. I'm not sure why I had to restart to trigger the rescan using the correct interface/address.

In real life probably it won't matter much. Either the scan finds the device straight away with default broadcast address, or otherwise it would require users to first go through installation, initial configuration, advanced network configuration and then come back to Add-on Store to scan again.

Reply:
One could consider adding a step to the setup wizard for the user to select the primary IP if there are multiple IP addresses detected on the system.